### DISCUSSION TITLE
on remote production instance - Cannot read properties of undefined (reading 'authStore')

### DISCUSSION THREAD

**Discussion by meltinbit** [pocketbase/pocketbase#2319] [Q&A]
Hi, I have a sveltekit project connected to Pocketbase.
When I connect to the local instance with new **PocketBase('http://127.0.0.1:8090')** everything is fine,
but when i try on a remote instance **new PocketBase('https://mydomain.com')** I get the error **Cannot read properties of undefined (reading 'authStore')**
The remote instance is normally accessible and on https.

Thanks in advance

**Comment by ganigeorgiev** [comment]
The base url shouldn't have effect on the `AuthStore` creation. It is used only when constructing the full url for the actual requests.

Could you provide a reproducible example or a code sample how do you initialize the SDKs?

**Comment by meltinbit** [comment]
I have a hooks.server.js with the following code

```js
import PocketBase from 'pocketbase';
import { serializeNonPOJOs } from '$lib/utils';

export const handle = async ({ event, resolve }) => {
	event.locals.pb = new PocketBase('http://127.0.0.1:8090');

	event.locals.pb.authStore.loadFromCookie(event.request.headers.get('cookie') || '');

	try {
		if (event.locals.pb.authStore.isValid) {
			await event.locals.pb.authStore.isValid && event.locals.pb.collection('users').authRefresh();
			event.locals.user = serializeNonPOJOs(event.locals.pb.authStore.baseModel);
		}
	} catch (err) {
		event.locals.pb.authStore.clear();
	}

	const response = await resolve(event);
	const isProd = process.env.NODE_ENV === 'production' ? true : false;
	response.headers.set(
		'set-cookie',
		event.locals.pb.authStore.exportToCookie({ secure: isProd, sameSite: 'Lax' })
	);
	

	return response;
};
```

**Reply by ganigeorgiev** [reply]
Are you sure that the error is triggered from a code within the hook and not from some other place where you may not have access to `event.locals.pb`?

**Reply by meltinbit** [reply]
here is the error i get in console...so it seems the line 8 

`event.locals.pb.authStore.loadFromCookie(event.request.headers.get("cookie") || "");`

TypeError: Cannot read properties of undefined (reading 'authStore')
    at Object.handle (/src/hooks.server.js:8:18)
    at Module.respond (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:243:40)
    at async file:///Users/fabiomangolini/Projects/Apps/SvelteKit/chatui/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:514:22

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce it with the template SvelteKit app.

The authStore is [initialized with the constructor](https://github.com/pocketbase/js-sdk/blob/master/src/Client.ts#L137) so `event.locals.pb.authStore` shouldn't be undefined.

What is the version of the JS SDK you are using?

**Reply by meltinbit** [reply]
0.13.1

i just tried now to update to 0.14.0 but nothing changed

**Reply by meltinbit** [reply]
I found the mistake, sorry but my fault...i initialized it wrong

`event.locals.pb = new PocketBase("http://127.0.0.1:8090");
	const pb = new PocketBase("https://mydomain.com");`

**Reply by meltinbit** [reply]
Anyway, I think Pocketbase is amazing, keep it up!
